dcspy/logitech.py

```diff
@@ -137,6 +141,28 @@ def _setup_plane_callback(self):
             dcsbios_buffer = getattr(import_module('dcspy.dcsbios'), ctrl.output.klass)
             dcsbios_buffer(parser=self.parser, callback=partial(self.plane.set_bios, ctrl_name), **ctrl.output.args.model_dump())
 
+    def gkey_callback_handler(self, gkeyOrButtonString: str, key: int, mode: int, keyDown: int) -> None:
+        """
+        Logitech G-Key callback handler.
+
+        Sent action to DCS-BIOS via network socket.
+
+        :param gkey: G-Key number
+        :param gkeyOrButtonString: G-Key string
+        :param context: context
+        """
+        LOG.debug(f'Button {gkeyOrButtonString} is pressed, key down: {keyDown}')
+        gkey = Gkey(key=key, mode=mode)
+        gkey_request = self.plane.button_request(gkey)
+        if gkey_request:
+            if self.plane.is_push_button(gkey):
+                request = f'{gkey_request} {keyDown}\n'
+                self.socket.sendto(bytes(request, 'utf-8'), SEND_ADDR)
+            elif not keyDown:
+                return
```

Comment:
We ignore any events for when the key is released if its not configured as a `push_button`. This is so we don't trigger further events for toggles, cycle buttons, etc.